src/documentation/io/testing.md

```diff
@@ -108,11 +108,465 @@ This allows the I/O transform's unit tests to act as if they have a perfect netw
 
 If your I/O transform allows batching of reads/writes, you must force the batching to occur in your test. Having configurable batch size options on your I/O transform allows that to happen easily. These must be marked as test only.
 
+
+## I/O Transform Integration Tests {#i-o-transform-integration-tests}
+
+
+### Goals  {#it-goals}
+
+*   Allow end to end testing of interactions between data stores, I/O transforms, and runners, simulating real world conditions.
+*   Allow both small scale and large scale testing.
+*   Self contained: require the least possible initial setup or existing outside state, besides the existence of a data store that the test can modify.
+*   Anyone can run the same set of I/O transform integration tests that Beam runs on its continuous integration servers.
+
+
+### Integration tests, data stores, and Kubernetes {#integration-tests-data-stores-and-kubernetes}
+
+In order to test I/O transforms in real world conditions, you must connect to a data store instance.
+
+The Beam community hosts the data stores used for integration tests in Kubernetes. In order for an integration test to be run in Beam's continuous integration environment, it must have Kubernetes scripts that set up an instance of the data store.
+
+However, when working locally, there is no requirement to use Kubernetes. All of the test infrastructure allows passing in connection info, so developers can use their preferred hosting infrastructure for local development.
+
+
+### Running integration tests {#running-integration-tests}
+
+The high level steps for running an integration test are:
+1.  Set up the data store corresponding to the test being run
+1.  Run the test, passing it connection info from the just created data store
+1.  Clean up the data store
+
+Since setting up data stores and running the tests involves a number of steps, and we wish to time these tests when running performance benchmarks, we use PerfKit Benchmarker (PKB) to manage the process end to end. With a single command, you can go from an empty Kubernetes cluster to a running integration test.
```

Comment:
would recommend removing (PKB) and using the full proper name PerfKit Benchmarker throughout